### ISSUE TITLE
SD write code causes random crashes

### ISSUE THREAD

**Issue by lifehackerhansol** [DS-Homebrew/TWiLightMenu#1691]
**Console Configuration:** DSi XL, 1.4.5, Unlaunch 2.0
**TWiLight Menu++ Version:** 23.0.1
**TWiLight Menu++ Theme:** DSi theme (though kind of irrelevant)

### Expected behaviour
ESRB rating works normally.

### Actual behaviour
ESRB rating sometimes causes crashes

### Steps to reproduce

1. Enable ESRB rating screen
2. Try to play a game with an ESRB rating
3. If it doesn't crash, just do it a few times, it'll eventually happen.

**Comment by NightScript370** [comment]
Can you upload your settings configuration file?

**Comment by RocketRobz** [comment]
I cannot reproduce this on my DSi, 3DS, and flashcard.

**Comment by lifehackerhansol** [comment]
@NightScript370 
```
[SRLOADER]
PAGE_NUMBER = 0
CURSOR_POSITION = 3
STARTMENU_CURSOR_POSITION = 0
CONSOLE_MODEL = 0
AUTORUNGAME = 0
WIFI_LED = 1
LANGUAGE_SET = 1
REGION_SET = 1
LANGUAGE = -1
TITLELANGUAGE = -1
USE_GBARUNNER2 = 1
SHOWLOGO = 0
SECONDARY_ACCESS = 0
SHOW_MAIN_MENU = 0
THEME = 0
SHOW_DIRECTORIES = 1
SHOW_BOX_ART = 2
ANIMATE_DSI_ICONS = 1
SYS_REGION = 1
LAUNCHER_APP = -1
SLOT1_LAUNCHMETHOD = 1
BOOTSTRAP_FILE = 0
USE_BOOTSTRAP = 1
AK_VIEWMODE = 2
AK_SCROLLSPEED = 4
AK_THEME = zelda
AK_ZOOM_ICONS = 1
SHOW_12H_CLOCK = 1
AUTORUNSLOT1 = 1
FRAME_RATE = 60
MACRO_MODE = 0
COLOR_MODE = 0
BLUE_LIGHT_FILTER_LEVEL = 0
DSIWARE_EXPLOIT = 0
GBAR2_DLDI_ACCESS = 0
SD_REMOVE_DETECT = 1
SHOW_MICROSD = 0
DSI_SPLASH = 2
DSI_SPLASH_AUTO_SKIP = 1
NINTENDO_LOGO_COLOR = 1
SHOW_SELECT_MENU = 1
SETTINGS_MUSIC = 2
DSI_MUSIC = 2
PHOTO_BOXART_COLOR_DEBAND = 1
SHOW_NDS = 1
SHOW_GBA = 2
SHOW_RVID = 1
SHOW_XEX = 1
SHOW_A26 = 1
SHOW_A52 = 1
SHOW_A78 = 1
SHOW_INT = 1
SHOW_NES = 1
SHOW_GB = 1
SHOW_SMSGG = 1
SHOW_MDGEN = 3
SHOW_SNES = 1
SHOW_PCE = 1
UPDATE_RECENTLY_PLAYED_LIST = 1
SORT_METHOD = 0
SHOW_HIDDEN = 0
PREVENT_ROM_DELETION = 0
FLASHCARD = 0
FC_SAVE_ON_SD = 0
SLOT1_ENABLESD = 0
SLOT1_SCFG_UNLOCK = 0
SLOT1_TOUCH_MODE = 0
LIMITED_MODE = 0
DSIWARE_BOOTER = 1
DSIWARE_TO_SD = 1
R4_THEME = unused
DSI_THEME = dark
3DS_THEME = light
GBA_BORDER = default.png
UNLAUNCH_BG = default.gif
UNLAUNCH_PATCH_REMOVE = 0
FONT = default
DS_CLASSIC_CUSTOM_FONT = 0
SMS_GG_IN_RAM = 0
WIDESCREEN = 0
ROM_FOLDER = sd:/
SECONDARY_ROM_FOLDER = fat:/
SECONDARY_PAGE_NUMBER = 0
SECONDARY_CURSOR_POSITION = 0
PREVIOUS_USED_DEVICE = 0
SLOT1_LAUNCHED = 0
LAUNCH_TYPE = 1
SECONDARY_LAUNCH_TYPE = 0
HOMEBREW_BOOTSTRAP = 1
HOMEBREW_HAS_WIDE = 0
DONT_SHOW_CLUSTER_WARNING = 0
ROM_PATH = sd:/SNEmulDS.nds

[NDS-BOOTSTRAP]
LANGUAGE = -1
REGION = -2
EXTENDED_MEMORY = 0
FORCE_SLEEP_PATCH = 0
SOUND_FREQ = 1
BOOST_CPU = 0
BOOST_VRAM = 0
DSI_MODE = 0
CARD_READ_DMA = 1
ASYNC_CARD_READ = 0
```

I'll retry this from other devices. Only tested the DSi thus far.

**Comment by lifehackerhansol** [comment]
Testing again, it doesn't happen.

Crash seems to randomly happen. It's happened about three times to me on this release alone, though.

**Comment by lifehackerhansol** [comment]
This now consistently crashes on Settings. Doesn't appear to happen anywhere else.

**Comment by lifehackerhansol** [comment]
Tried Zoo Keeper and it booted the game fine.

**Comment by lifehackerhansol** [comment]
Tried a few other games (Bomberman, Pokemon White 1) and those ran as well, showed the ESRB rating and all.

**Comment by RocketRobz** [comment]
You might need to launch them around 10 times or more to reproduce the crash.
If it doesn't crash, idk what's going on.

**Comment by lifehackerhansol** [comment]
Ooookay yeah third boot on Bomberman and it crashed as you said.

**Comment by lifehackerhansol** [comment]
This is also reproducible on a flashcart. Seems to be issue with the whole thing.

**Comment by sks316** [comment]
Ran into this issue today with the latest update, it's definitely not an isolated issue

**Comment by RocketRobz** [comment]
The crashing issue in Settings and Manual are fixed by e6a71415 & 9be71be4, though not related to this issue.

**Comment by RocketRobz** [comment]
As of later version, this seems to be fixed, as I've heard no recent reports about freezing after first-time setup, and I've also encountered no crashes when launching games with the ESRB Splash enabled.
